chia/util/streamable.py

```diff
@@ -134,54 +120,38 @@ def dataclass_from_dict(klass: Type[Any], d: Any) -> Any:
         return klass(d)
 
 
-@overload
-def recurse_jsonify(d: Union[List[Any], Tuple[Any, ...]]) -> List[Any]:
-    ...
-
-
-@overload
-def recurse_jsonify(d: Dict[str, Any]) -> Dict[str, Any]:
-    ...
-
-
-def recurse_jsonify(d: Union[List[Any], Tuple[Any, ...], Dict[str, Any]]) -> Union[List[Any], Dict[str, Any]]:
+def recurse_jsonify(d: Any) -> Any:
     """
     Makes bytes objects and unhashable types into strings with 0x, and makes large ints into
     strings.
     """
-    if isinstance(d, list) or isinstance(d, tuple):
+    if dataclasses.is_dataclass(d):
```

Comment:
Should this work for any dataclass?  Or should this be a streamable thing.  If streamable we can use a streamable hint in a new overload above and a streamable function to get the attributes.

Reply:
in a way this is not a publicly facing function, and we control all call sites; so we can make it do whatever we want really. For example, the `Streamable` serialization format doesn't support enums, so we could probably remove that here (but it would take some work to ensure we don't rely on it in some way that isn't covered by a test).

The important think I'm trying to make this function do is to support types that do *not* derive from `Streamable`,  but implement a protocol that can more easily be implemented, by a rust native type for instance. I'm not sure if it's possible with any python bindings for a native type to derive from a pure python type. It's not supported by pyo3 that that we use in rust, and I think in other binding libraries it would be very complicated.

The hook I'm envisioning is that types that aren't a dataclass (nor one of the built-in Dict, List, int, str, None) can implement a `to_json_dict()` to do the same thing.

In other words, I'm trying to separate this from the `Streamable` class.